plugins/provisioners/ansible/plugin.rb

```diff
@@ -3,21 +3,73 @@
 module VagrantPlugins
   module Ansible
     class Plugin < Vagrant.plugin("2")
+
       name "ansible"
       description <<-DESC
-      Provides support for provisioning your virtual machines with
-      Ansible playbooks.
+      Provides support for provisioning your virtual machines with Ansible
+      from the Vagrant host (`ansible`) or from the guests (`ansible_local`).
       DESC
 
-      config(:ansible, :provisioner) do
-        require File.expand_path("../config", __FILE__)
-        Config
+      config("ansible", :provisioner) do
+        require_relative "config/host"
+        Config::Host
+      end
+
+      config("ansible_local", :provisioner) do
```

Comment:
The use of the term _local_ in `ansible_local` is consistent with [`local-exec`](https://docs.vagrantup.com/v2/push/local-exec.html) for `vagrant push`. (But obviously the naming schemes differ in other ways, such as `-` vs `_` and whether `local` comes first or second.)